src/main/java/com/squareup/javapoet/ParameterizedTypeName.java

```diff
@@ -56,6 +56,10 @@
     return new ParameterizedTypeName(rawType, typeArguments);
   }
 
+  @Override StringBuilder binary(StringBuilder stringBuilder) {
+    return rawType.binary(stringBuilder);
```

Comment:
May be not, but it doesn't hurt. It mimics this assertion:

`assertEquals("java.util.HashSet", new HashSet<Byte>().getClass().getName());`

Yes, `getClass()` discards the type variable here. On the JavaPoet side, you'd need to check for "t instanceof ParameterizedTypeName" and call `t.rawType.binary()`. This is just a shortcut for it.